csharp/ql/src/semmle/code/csharp/PrintAst.qll

```diff
@@ -100,15 +100,16 @@ private ValueOrRefType getAnInterestingBaseType(ValueOrRefType type) {
   not type instanceof ArrayType and
   not type instanceof NullableType and
   result = type.getABaseType() and
-  isInterestingBaseType(result)
+  isInterestingBaseType(type, result)
 }
 
-private predicate isInterestingBaseType(ValueOrRefType base) {
+private predicate isInterestingBaseType(ValueOrRefType type, ValueOrRefType base) {
   not base instanceof ObjectType and
   not base.getQualifiedName() = "System.ValueType" and
   not base.getQualifiedName() = "System.Delegate" and
   not base.getQualifiedName() = "System.MulticastDelegate" and
-  not base.getQualifiedName() = "System.Enum"
+  not base.getQualifiedName() = "System.Enum" and
+  exists(TypeMention tm | tm.getTarget() = type and tm.getType() = base)
```

Comment:
I'm not sure what would be the best here. This is definitely a fix for the time being, because this way there are no empty `(Base types)` nodes in the tree. However, the enduser might actually want to see the base types of a declaration, and not just the type mentions of the base types. But we're not creating type mentions for these "implicit" base types of `record` declarations.

Reply:
I think that is fine for now.